mkosi/__init__.py

```diff
@@ -1642,34 +1645,63 @@ def clean_dnf_metadata(root: str) -> None:
     keeping the dnf metadata, since it's not usable from within the
     image anyway.
     """
-    dnf_path = root + "/bin/dnf"
+    dnf_metadata_paths = [
+        os.path.join(root, "var/lib/dnf"),
+        *glob.glob(f"{os.path.join(root, 'var/log/dnf')}.*"),
+        *glob.glob(f"{os.path.join(root, 'var/log/hawkey')}.*"),
+        os.path.join(root, "var/cache/dnf"),
+    ]
+    dnf_path = os.path.join(root, "bin/dnf")
     keep_dnf_data = os.access(dnf_path, os.F_OK, follow_symlinks=False)
 
-    if not keep_dnf_data:
-        MkosiPrinter.print_step("Cleaning dnf metadata...")
-        remove_glob(
-            root + "/var/lib/dnf", root + "/var/log/dnf.*", root + "/var/log/hawkey.*", root + "/var/cache/dnf"
-        )
+    if not keep_dnf_data or all(not os.path.exists(path) for path in dnf_metadata_paths):
```

Comment:
The check here is only to bail out and not print the "Cleaning dnf metadata" message, if there is none, right?

It's maybe an unnecessarily picky, but since this repeats in the other functions, i'd factor it out and name it something like `none_of_exists(path: Iterable) -> bool`

Reply:
Let's leave this for a subsequent cleanup if it becomes a problem.